enterprise/middleware.py

```diff
@@ -0,0 +1,28 @@
+from enterprise.models import EnterpriseCustomer, UserDataSharingConsentAudit
+from enterprise.tpa_pipeline import (
+    active_provider_enforces_data_sharing,
+    get_consent_status_for_request,
+    get_enterprise_customer_for_request
+)
+class ResetSessionIfPipelineBrokenMiddleware(object):
+    """
+    Middleware signs the user out if they need to provide data sharing consent,
```

Comment:
Based on my understanding of the pipeline, this cannot be done without middleware. The pipeline is entirely "opt-in"; it consists of state stored in the session. That state doesn't do anything unless it's interpreted by a view or form that understands it. If the user browses to a page outside the pipeline (meaning a page that doesn't eventually resolve to a redirect to re-enter the pipeline), then we can't "grab" the user to bring them back.